### PR TITLE
feat: 719 - Suggestion Manager with suggestions for the latest input

### PR DESCRIPTION
**Pull request by monsieurtanuki**
New files:
* `api_suggestion_manager_test.dart`: Unit tests for `SuggestionManager`
* `suggestion_manager.dart`: Manager that returns the suggestions for the latest input.

Impacted files:
* `open_food_api_client.dart`: added a comment mentioning `SuggestionManager`.
* `openfoodfacts.dart`: exported new file `suggestion_manager.dart`

### What
- New class `SuggestionManager` that was needed for autocomplete widgets.
- We already had a method in order to get suggestions.
- With the manager, the user sees the most relevant suggestions (that match the latest input), regardless of the time taken by each individual call to the `getSuggestions` method (typically, when the oldest call takes more time than a more recent call).

### Fixes bug(s)
- Closes: #719

### CODE REVIEW

File: lib/src/open_food_api_client.dart [hunk 1]
```diff
@@ -1131,10 +1143,12 @@ class OpenFoodAPIClient {
       queryType: queryType,
       addCredentialsToBody: false,
     );
-    return json.decode(response.body);
+    return HttpHelper().jsonDecode(response.body);
   }
 
   /// cf. https://openfoodfacts.github.io/openfoodfacts-server/reference/api-v3/#get-/api/v3/taxonomy_suggestions
+  ///
+  /// Consider using [SuggestionManager].
```
Comment (M123-dev) [inline lib/src/open_food_api_client.dart:1151]: Could you update the README

File: test/api_get_suggestions_test.dart [hunk 2]
```diff
@@ -257,7 +260,7 @@ void main() {
     test('Suggestions for additives', () async {
       List<String> result = await OpenFoodAPIClient.getSuggestions(
         TagType.ADDITIVES,
-        language: OpenFoodFactsLanguage.RUSSIAN,
```
Comment (M123-dev) [inline test/api_get_suggestions_test.dart:260]: What impact does this have

**Review by M123-dev — APPROVED** [review]
Heyy @monsieurtanuki looks great besides some missing documentation in the README


Also, sorry for my late response, but now I'm done with exams for some time at least :D

File: test/api_get_suggestions_test.dart (hunk 2 above)
Comment (monsieurtanuki) [reply test/api_get_suggestions_test.dart:260]: For some reason the test does not pass anymore in Russian.
As we more or less agreed that it was more relevant to run integration tests with server data than unit tests with local json files, once in a while we have to "fix" tests.

File: lib/src/open_food_api_client.dart (hunk 1 above)
Comment (monsieurtanuki) [reply lib/src/open_food_api_client.dart:1151]: > Could you update the README

I will once the code is merged, as I cannot point to the doc of a class that does not exist yet.

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for your review!

**Comment by monsieurtanuki** [comment]
@M123-dev When is the doc refreshed btw?

**Pull request merged**